rig/machine_control/machine_controller.py

```diff
@@ -307,6 +332,63 @@ def boot(self, width, height,
         return True
 
     @ContextMixin.use_contextual_arguments()
+    def discover_connections(self, x=0, y=0):
```

Comment:
@mossblaser - do you have any feelings about this method being called by `get_system_info`? I realised that I've not been calling this method when it would probably be in my interests to do so.